test/integration/serverTest.js

```diff
@@ -823,4 +855,80 @@ describe('Brave Sensor server', () => {
       })
     })
   })
+
+  describe('POST request heartbeat events with mock INS Firmware State Machine', () => {
+    beforeEach(async () => {
+      await redis.clearKeys()
+      await db.clearSessions()
+      await db.clearLocations()
+      await db.clearClients()
+
+      const client = await clientFactory(db)
+      await db.createLocation(
+        testLocation1Id,
+        testLocation1PhoneNumber,
+        MOVEMENT_THRESHOLD,
+        STILLNESS_TIMER,
+        DURATION_TIMER,
+        1000,
+        INITIAL_TIMER,
+        ['+15005550006'],
+        '+15005550006',
+        ['+15005550006'],
+        1000,
+        'locationName',
+        radar_coreID,
+        radar_coreID,
+        'Innosent',
+        'alertApiKey',
+        true,
+        true,
+        firstLowBatteryAlert,
+        client.id,
+      )
+      sandbox.stub(braveAlerter, 'startAlertSession')
+      sandbox.stub(braveAlerter, 'sendSingleAlert')
+    })
+
+    afterEach(async () => {
+      await redis.clearKeys()
+      await db.clearSessions()
+      await db.clearLocations()
+      await db.clearClients()
+      sandbox.restore()
+      helpers.log('\n')
+    })
+
+    it('should return 200 for a heartbeat request with no headers', async () => {
+      const response = await chai.request(server).post('/api/heartbeat').send({})
+      expect(response).to.have.status(200)
+    })
+
+    it('should return 200 for a valid heartbeat request', async () => {
+      const request = {
+        coreid: radar_coreID,
+        data: `{"doorMissedMsg": 0, "doorLowBatt": true, "doorLastHeartbeat": 1000, "resetReason": "NONE", "states":[]}`,
+      }
+      const response = await chai.request(server).post('/api/heartbeat').send(request)
+      expect(response).to.have.status(200)
+    })
+
+    it('should not update sentLowBatteryAlertAt if battery is nominal', async () => {
+      const testLocation = await db.getLocationData(testLocation1Id)
+      await nominalHeartbeat(radar_coreID)
+      const locationAfterHeartbeat = await db.getLocationData(testLocation1Id)
+      expect(locationAfterHeartbeat.sentLowBatteryAlertAt).to.deep.equal(testLocation.sentLowBatteryAlertAt)
+    })
+
+    it('should update sentLowBatteryAlertAt if battery is low and only when not timed out', async () => {
+      const testLocation = await db.getLocationData(testLocation1Id)
+      await lowBatteryHeartbeat(radar_coreID)
+      const locationAfterHeartbeat = await db.getLocationData(testLocation1Id)
+      expect(locationAfterHeartbeat.sentLowBatteryAlertAt).to.not.equal(testLocation.sentLowBatteryAlertAt)
+
+      await lowBatteryHeartbeat(radar_coreID)
+      const locationAfterTwoHeartbeats = await db.getLocationData(testLocation1Id)
+      expect(locationAfterTwoHeartbeats.sentLowBatteryAlertAt).to.deep.equal(locationAfterHeartbeat.sentLowBatteryAlertAt)
+    })
```

Comment:
I think that there is something wrong in your code. Because you don't have `LOW_BATTERY_ALERT_TIMEOUT_THRESHOLD_TEST` in your `.travis.yaml` file yet, the value of the threshold is `undefined`. So how is the check whether it has passed the threshold or not working?

I think that any boolean expression compared with `undefined` will give you `false`. So `lowBatteryAlertTimeout` is probably returning `false` all the time in Travis (and in your local if you've been running your tests without `LOW_BATTERY_ALERT_TIMEOUT_THRESHOLD_TEST` in your `.env` file.